lib/web3/contract.js

```diff
@@ -57,7 +56,7 @@ var addFunctionsToContract = function (contract, abi) {
     abi.filter(function (json) {
         return json.type === 'function';
     }).map(function (json) {
-        return new SolidityFunction(json, contract.address);
+        return new SolidityFunction(contract._web3, json, contract.address);
```

Comment:
Also renaming is not a part of this pr. If we decide to rename all of this, I suggest to move all abi-related stuff to it's own library. I initially wanted to this with [solidity.js](https://github.com/ethereum/solidity.js), but was stopped by more urgent things.